src/main/java/io/knowledgeassets/myskills/server/community/query/CommunityQueryController.java

```diff
@@ -45,10 +47,10 @@ public CommunityQueryController(CommunityQueryService communityQueryService,
 	})
 	@PreAuthorize("isAuthenticated()")
 	@GetMapping(path = "/communities", produces = MediaType.APPLICATION_JSON_VALUE)
-	public List<CommunityResponse> getCommunities() {
+	public List<RecommendedCommunityResponse> getCommunities() {
 		final User currentUser = currentUserService.getCurrentUser();
-		return communityQueryService.getCommunities()
-				.map(convertCommunityToCommunityResponse(currentUser))
+		return communityQueryService.getCommunitiesRecommendedForUser(currentUser.getId())
+				.map(convertRecommendedCommunityToRecommendedCommunityResponse(currentUser))
```

Comment:
I think the list API for communities should always return the simple representation because we don't want to display the members in the list view of communities, only the managers.

Reply:
The only reason to return a list of members for a community in the list now is that the web app uses the member list [to determine if a user is a member of the community](https://github.com/KnowledgeAssets/myskills-webapp/blob/master/src/app/communities/communities.component.ts#L106) or not (to display / hide `Join` button).
So there has to be another way to decide if a user is a member of a community.

Reply:
Yes, I know, I've seen that implementation in the web app. I've made a proposal to implement an API endpoint `GET /users/{userId}/communities` which returns the communities the user is a member of. In my point of view, this would be the cleaner approach to determine the desired information instead of scanning the members list of communities.

Reply:
Alright, let's keep that logic for the moment. I see that changing this implies rework of the communities API in general and the way how the web app uses the API.